portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/ClientJS.testcase

```diff
@@ -0,0 +1,91 @@
+definition {
+
+	property analytics.cloud.release = "true";
+	property analytics.cloud.upstream = "true";
+	property portal.release = "false";
+	property portal.upstream = "true";
+	property property.group = "analytics.cloud.profile";
+	property testray.main.component.name = "Analytics Cloud";
+
+	setUp {
+		task ("Copy the token from AC") {
+			ACUtils.launchAC();
+
+			ACSettings.takeTokenFromAnalyticsCloud();
+		}
+
+		task ("Set up instance and sign in DXP") {
+			TestCase.setUpPortalInstance();
+
+			ACUtils.launchDXP();
+		}
+
+		task ("Add a new site and page") {
+			ACUtils.addSiteAndPage();
+		}
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		ACUtils.tearDownDXP();
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			JSONGroup.deleteGroupByName(groupName = "Site Name");
+
+			CustomFields.tearDownCP();
+
+			JSONUser.tearDownNonAdminUsers();
+		}
+
+		ACUtils.tearDownAC();
```

Comment:
I'm currently working on fixing teardowns and one thing I noticed to help our teardowns is removing the `JSONUser.tearDownNonAdminUsers` as well as moving the `ACUtils.tearDownAC` to before the `ACUtils.tearDownDXP`. Can you make those changes since this is a new file? Thanks!

I'll hopefully send over the PR soon with an explanation for those changes.

Reply:
Yes, of course. But I have a question related to removing the `JSONUser.tearDownNonAdminUsers` we have a lot of tests that create new users, if we remove this line how these users will be erased?

Reply:
It's erased during `PortalInstances.tearDownCP();` macro. During our teardown step all of our runs go through that macro which the step already does. In your instance it looks like it wouldn't affect anything since it is part of the else statement so it wouldn't run together. But I don't believe we have an instance where the else statement ever runs

Reply:
Thanks for the explanation Josh :male_detective: